src/rest/class-wp-rest-abilities-list-controller.php

```diff
@@ -0,0 +1,292 @@
+<?php declare( strict_types = 1 );
+
+/**
+ * REST API: WP_REST_Abilities_List_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 0.1.0
+ */
+
+/**
+ * Core controller used to access abilities via the REST API.
+ *
+ * @since 0.1.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Abilities_List_Controller extends WP_REST_Controller {
+
+	/**
+	 * REST API namespace.
+	 *
+	 * @since 0.1.0
+	 * @var string
+	 */
+	protected $namespace = 'wp/v2';
+
+	/**
+	 * REST API base route.
+	 *
+	 * @since 0.1.0
+	 * @var string
+	 */
+	protected $rest_base = 'abilities';
+
+	/**
+	 * Registers the routes for abilities.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[a-zA-Z0-9\-\/]+)',
+			array(
+				'args' => array(
+					'id' => array(
+						'description' => __( 'Unique identifier for the ability.' ),
+						'type'        => 'string',
+						'pattern'     => '^[a-zA-Z0-9\-\/]+$',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
```

Comment:
I see that `get_items_permissions_check` is used for both routes, so maybe renaming it to something more general like `get_permissions_check` would work better here.